Quick Start V2: update checklist

danielebogo opened the pull request:
Refs. #10860 

This PR updates the current quick start checklist. It uses 2 sections to divide the completed tasks from the todo tasks.

![qs-section](https://user-images.githubusercontent.com/912252/51679828-e8762f80-1fd7-11e9-8e54-eb62b0531347.gif)

Because this update is behind feature flag, I created a new data flow _QuickStartChecklistManager_ to handle the table view datasource and delegate.

@SylvesterWilmott this is the final result:

![simulator screen shot - iphone x - 2019-01-24 at 13 52 16](https://user-images.githubusercontent.com/912252/51682542-91745880-1fdf-11e9-9018-e8ae9225fd52.png)

![simulator screen shot - iphone x - 2019-01-24 at 13 52 18](https://user-images.githubusercontent.com/912252/51682552-96d1a300-1fdf-11e9-97f8-8eba25cb986b.png)


**To test:**
- Create a new site and turn on quick start (or put return true in shouldShowQuickStartChecklist())
- Select _Customize Your Site_ to complete at least one task
- You should see the todo list separated from the completed list
- Tap on the _Complete_ header to expand/hide the section

SylvesterWilmott commented:
This looks fantastic @danielebogo thanks!

There seems to be an issue with the separators:
- The separators don't align with the text
- The separator on the first item in the completed list is full width

danielebogo commented:
@SylvesterWilmott I'm fixing it!

danielebogo commented:
@SylvesterWilmott here:

![simulator screen shot - iphone x - 2019-01-24 at 17 41 50](https://user-images.githubusercontent.com/912252/51697315-6ef23780-1fff-11e9-87d1-0bf77c84a094.png)

nheagy commented on WordPress/Classes/ViewRelated/Blog/QuickStartChecklistManager.swift:
```diff
@@ -0,0 +1,178 @@
+        let completed = (blog.completedQuickStartTours ?? []).map { $0.tourID }
+        let skipped = (blog.skippedQuickStartTours ?? []).map { $0.tourID }
+
+        completedToursKeys = Set(completed).union(Set(skipped))
```
I think there is a logic problem here. `blog.skippedQuickStartTours` are tours that have been skipped from the notices. This isn't supposed to mark them as complete. This is different than the `Skip` button(s) in the UI, which _do_ mark them as complete. That's a difference between what "skip" means in the code vs. the UI. 

In the new UI, the sideways UI will mark them as complete, as for all our purposes they will be. Let me know if that doesn't make sense.

SylvesterWilmott replied:
Could there be a benefit of making a distinction between skipped and completed for any future iterations we might make?

danielebogo commented on WordPress/Classes/ViewRelated/Blog/QuickStartChecklistManager.swift:
```diff
@@ -0,0 +1,178 @@
+}
+
+private enum Sections: Int, CaseIterable {
+    static let footerHeight = CGFloat(20.0)
```
This is actually the footer view height of the first section 😊

SylvesterWilmott commented:
@danielebogo let's just change the stroke colour to `greyLighten20`

SylvesterWilmott commented:
Thanks @danielebogo !

danielebogo commented on WordPress/Classes/ViewRelated/Blog/QuickStartChecklistViewController.swift:
```diff
@@ -1,21 +1,45 @@
             self.tableView?.dataSource = dataSource
         }
     }
+    private var dataManager: QuickStartChecklistManager? {
```
Ok

danielebogo commented:
@nheagy I fixed everything and duplicate the `QuickStartChecklistViewController`. I also changed the way this new vc is instantiated. It uses an enum (easy to use by objc classes) rather than injecting the configuration structs.

danielebogo commented:
Thanks @nheagy @SylvesterWilmott
